### PR TITLE
fix: Other user profile menu flickering

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

### Issues
In OtherUserProfile screen: on blocking/unblocking user bottomSheet menu was flickering as crazy  (see the attached video)

### Causes (Optional)
Problem was that coroutine that is fetching the userData was creating a separate coroutine for fetching menu data. 
So everytime userData was changed a new flow for fetching menuData was created and they were fighting to set menu content.

### Solutions

Make fetching menuData a separate Flow.

### Attachments (Optional)

https://user-images.githubusercontent.com/6539347/208936320-93581357-d448-4967-90f8-598d3dfc431c.mp4

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1253/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5266848)(5266848)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5266848)(5266848)-dev-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt [hunk 1]
```diff
@@ -144,76 +148,87 @@ class OtherUserProfileScreenViewModel @Inject constructor(
...
+                                    userName = mapUserLabel(otherUser),
+                                    teamName = userResult.team?.name.orEmpty(),
+                                    email = otherUser.email.orEmpty(),
+                                    phone = otherUser.phone.orEmpty(),
+                                    connectionState = otherUser.connectionStatus,
+                                    membership = userTypeMapper.toMembership(otherUser.userType),
+                                    groupState = groupInfo,
+                                    botService = otherUser.botService,
+                                )
+                            }
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt]: Maybe extract to a separate method to increase readability?

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt [hunk 2]
```diff
@@ -144,76 +148,87 @@ class OtherUserProfileScreenViewModel @Inject constructor(
...
+                            conversationTypeDetail = ConversationTypeDetail.Private(
+                                userAvatarAsset,
+                                userId,
+                                otherUser.BlockState
+                            ),
+                            isTeamConversation = conversationResult.conversation.isTeamGroup(),
+                            selfRole = Conversation.Member.Role.Member
+                        )
+                    )
+                }
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt]: Maybe also to extract to a separate method?

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1253/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5311003)(5311003)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5311003)(5311003)-dev-compat.apk)

**Review by gongracr — APPROVED** [review]
Looks good to me :) Although the challenge now is making classes and files more readable the bigger they grow 😅

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1253/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5317045)(5317045)-dev-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/compat/com.wire.android-vv0.0.1(5317045)(5317045)-dev-compat.apk)

**Pull request merged**